### PR TITLE
Modified README in examples folder

### PR DESCRIPTION
**Pull request by saloneerege**
Made changes to use variables in main.tf file and added defitions for those variables in variables.tf.

### CODE REVIEW

**Comment by saloneerege** [comment]
Pull request title appended with WIP since I am testing my changes. This branch should not be merged until then.

File: examples/README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
-# Example : Create SDDC
+# Provision an SDDC Example
+
+This is an example that supports SDDC management actions like creating an SDDC ,
```
Comment (shylasrinivas) [inline examples/README.md]: space before comma needs to be removed for SDDC

File: examples/variables.tf [hunk 2]
```diff
@@ -1,8 +1,24 @@
+variable "api_token"{
+  description = "API token used to auntheicate when calling the VMware Cloud Services API."
```
Comment (shylasrinivas) [inline examples/variables.tf]: fix spelling for authenticate

**Review by shylasrinivas — COMMENTED** [review]
do we want to also move num_host to variables.tf? It is set to 3 currently..

File: examples/README.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
-# Example : Create SDDC
+# Provision an SDDC Example
+
+This is an example that supports SDDC management actions like creating an SDDC, 
+updating or deleting an existing SDDC.
+
+To run the example:
+* Generate an API token in the [VMware Cloud on AWS console] (https://vmc.vmware.com/console/)
+* Update the `variables.tf` according to your test environment.
```
Comment (rchadalawada) [inline examples/README.md]: For completeness, please add the terraform commands also.

File: examples/main.tf [hunk 4]
```diff
@@ -16,10 +16,11 @@ data "vmc_customer_subnets" "my_subnets" {
...
 
+
 resource "vmc_sddc" "sddc_1" {
   org_id = data.vmc_org.my_org.id
 
-  sddc_name           = ""
+  sddc_name           = var.sddc_name
   vpc_cidr            = var.vpc_cidr
   num_host            = 3
   provider_type       = "AWS"
```
Comment (rchadalawada) [inline examples/main.tf:26]: what is the use of this parameter? Why is this set to AWS?

**Review by rchadalawada — CHANGES_REQUESTED** [review]

File: examples/main.tf (hunk 4 above)
Comment (saloneerege) [reply examples/main.tf:26]: The provider type determines what additional properties are available based on cloud provider (description from the VMC API explorer). This is the same parameter that can be changed to ZEROCLOUD

File: examples/README.md (hunk 3 above)
Comment (saloneerege) [reply examples/README.md]: Done

File: vmc/resource_vmc_sddc.go [hunk 5]
```diff
@@ -290,6 +294,7 @@ func resourceSddcRead(d *schema.ResourceData, m interface{}) error {
 		d.Set("vc_url", sddc.ResourceConfig.VcUrl)
 		d.Set("cloud_username", sddc.ResourceConfig.CloudUsername)
 		d.Set("cloud_password", sddc.ResourceConfig.CloudPassword)
+		d.Set("nsxt_reverse_proxy_url",sddc.ResourceConfig.NsxApiPublicEndpointUrl)
```
Comment (shylasrinivas) [inline vmc/resource_vmc_sddc.go:297]: Is your local branch synced with master? I think I saw nsxt related changes in another PR which was merged earlier today.

**Review by rchadalawada — CHANGES_REQUESTED** [review]
The READMEs still need some reorg. Lets discuss tomorrow

**Review by rchadalawada — COMMENTED** [review]
You may have to rebase your branch .. I see a lot more changes than just the README

**Comment by saloneerege** [comment]
> You may have to rebase your branch .. I see a lot more changes than just the README

@rchadalawada I rebased my branch with master and the changes other than the README are from the rebase.

File: README.md [hunk 6]
```diff
@@ -11,6 +11,51 @@ Requirements
 - [Terraform](https://www.terraform.io/downloads.html) 0.12+
 - [Go](https://golang.org/doc/install) 1.13 (to build the provider plugin)
 
+## Build the Provider
```
Comment (shylasrinivas) [inline README.md:14]: From the sample README @ https://github.com/terraform-providers/terraform-provider-null, this should be 'Building the Provider'.

File: examples/README.md [hunk 7]
```diff
@@ -1,3 +1,98 @@
...
 
-This example provisions a SDDC.
+This is an example that supports SDDC management actions like creating an SDDC, 
+updating or deleting an existing SDDC.
+
+To run the example:
+
+* Generate an API token using [VMware Cloud on AWS console] (https://vmc.vmware.com/console/)
+
+* Update the variables required parameters api_token and org_id in [variables.tf](https://github.com/vmware/terraform-provider-vmc/blob/master/examples/variables.tf) with your infrastructure settings.
```
Comment (shylasrinivas) [inline examples/README.md:10]: remove variables from the beginning of the sentence? - change to 'Update required parameters..'

File: examples/README.md [hunk 8]
```diff
@@ -11,9 +11,88 @@ To run the example:
...
+```sh 
   terraform apply \
   -var="api_token=xxxx" \
   -var="org_id=xxxx"
-  ```
+```
 
+# Using the provider
+
+The following instructions assume that the required parameters have updated in [variables.tf](https://github.com/vmware/terraform-provider-vmc/blob/master/examples/variables.tf) with your infrastructure settings.
```
Comment (rchadalawada) [inline examples/README.md:22]: Please correct the sentence.  -- "have been updated" or "are updated"

**Review by shylasrinivas — COMMENTED** [review]
Can you run linter check on the code?

**Pull request closed without merging**